scripts/parse_tools/parse_checkers.py

```diff
@@ -101,7 +101,13 @@ def check_dimensions(test_val, prop_dict, error, max_len=0):
     >>> check_dimensions("hi_mom", None, True) #doctest: +IGNORE_EXCEPTION_DETAIL
     Traceback (most recent call last):
     CCPPError: 'hi_mom' is invalid; not a list
+    >>> check_dimensions(["1:dim1", "dim2name"], None, True) #doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    CCPPError: '1:dim1 is an invalid dimension name; integer dimension indices not supported
+    >>> check_dimensions(["ccpp_constant_one:1", "dim2name"], None, True)
+    ['ccpp_constant_one:1', 'dim2name']
```

Comment:
Why is this okay? Isn't `1` an integer dimension index?

Reply:
a valid question! The problem I ran into is that sometimes the parse checker is called after the "ccpp_constant_one:" is prepended to the integer dimension (which I believe we're allowing) and sometimes it's called before.

So, the metadata could specify `dimensions = (2)` and we'd end up calling check_dimensions on `ccpp_constant_one:2`

I thought about diving deeper into figuring out what the specifics were of when we're calling what, but decided that `ccpp_constant_one:1` is the same as saying `1` and left it at that.

Reply:
Here is what is confusing: If "integer dimension indices not supported", how is the `1`, an integer okay? Also, when did we decide it is okay to use integers for array extents?

Reply:
basically, it's only ok if you do `ccpp_constant_one:1`. any other use of integer indices isn't ok. And that's because of my earlier comment - to appease the checker.

As for when we started allowing integer sizes, i believe @dustinswales brought that in recently for his work in SCM - https://github.com/NCAR/ccpp-framework/pull/652